docker/cygnus-ngsi/Dockerfile

```diff
@@ -0,0 +1,111 @@
+#
+# Copyright 2016 Telefonica Investigación y Desarrollo, S.A.U
+#
+# This file is part of fiware-cygnus (FI-WARE project).
+#
+# fiware-cygnus is free software: you can redistribute it and/or modify it under the terms of the GNU Affero
+# General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your
+# option) any later version.
+# fiware-cygnus is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the
+# implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License
+# for more details.
+#
+# You should have received a copy of the GNU Affero General Public License along with fiware-cygnus. If not, see
+# http://www.gnu.org/licenses/.
+#
+# For those usages not covered by the GNU Affero General Public License please contact with iot_support at tid dot es
+#
+
+FROM centos:6
+MAINTAINER Francisco Romero Bueno <francisco.romerobueno@telefonica.com>
+
+ENV CYGNUS_USER "cygnus"
+ENV CYGNUS_HOME "/opt/fiware-cygnus"
+ENV CYGNUS_VERSION "0.13.0_SNAPSHOT"
+
+ENV GIT_URL_CYGNUS https://github.com/telefonicaid/fiware-cygnus.git
+ENV GIT_REV_CYGNUS develop
+
+ENV MVN_VER "3.2.5"
+ENV MVN_TGZ "apache-maven-${MVN_VER}-bin.tar.gz"
+ENV MVN_URL "http://www.eu.apache.org/dist/maven/maven-3/${MVN_VER}/binaries/${MVN_TGZ}"
+ENV MVN_HOME "/opt/apache-maven"
+
+ENV FLUME_VER "1.4.0"
+ENV FLUME_TGZ "apache-flume-${FLUME_VER}-bin.tar.gz"
+ENV FLUME_URL "http://archive.apache.org/dist/flume/${FLUME_VER}/${FLUME_TGZ}"
+ENV FLUME_HOME "/opt/apache-flume"
+
+ENV CONF_PATH "/opt/apache-flume/conf"
+ENV CONF_FILE "/opt/apache-flume/conf/agent.conf"
+ENV AGENT_NAME "cygnus-ngsi"
+ENV LOG_LEVEL "INFO"
+ENV LOG_APPENDER "console"
+
+ENV JAVA_HOME /usr/lib/jvm/java-1.6.0
+WORKDIR /opt
+
+RUN adduser --comment "${CYGNUS_USER}" ${CYGNUS_USER}
+
+# Install dependencies (Java, Maven, Flume)
+RUN yum -y install tar nc which git java-1.6.0-openjdk-devel && \
+    yum clean all && \
+    curl --remote-name --location --insecure --silent --show-error "${MVN_URL}" && \
+    tar xzvf ${MVN_TGZ} && \
+    mv apache-maven-${MVN_VER} ${MVN_HOME} && \
+    export MAVEN_OPTS="-Xmx512m -XX:MaxPermSize=128m" && \
+    curl --remote-name --location --insecure --silent --show-error "${FLUME_URL}" && \
+    tar zxf ${FLUME_TGZ} && \
+    mv apache-flume-${FLUME_VER}-bin ${FLUME_HOME} && \
+    mkdir -p ${FLUME_HOME}/plugins.d/cygnus && \
+    mkdir -p ${FLUME_HOME}/plugins.d/cygnus/lib && \
+    mkdir -p ${FLUME_HOME}/plugins.d/cygnus/libext && \
+    chown -R cygnus:cygnus ${FLUME_HOME}
+
+# Create Cygnus log folder
+RUN mkdir ${FLUME_HOME}/log
+
+# Clone fiware-cygnus
+RUN git clone ${GIT_URL_CYGNUS} ${CYGNUS_HOME} && \
```

Comment:
By doing a `swallow clone` with `--depth 1` you can decrease the image size in more than 100 MB